src/Type.ml

```diff
@@ -1,23 +1,36 @@
 open Exceptions
+open Core
 
 type t = INT
        | BOOL
        | CHAR
        | STRING
        | LIST
-       [@@deriving sexp]
+       | TVAR of string
+       | ARRAY of (t * t)
+       [@@deriving compare,sexp]
 
-let of_string : string -> t = function
+let of_atomic_string (s:string) : t =
+  match s with 
   | "Int"    -> INT
   | "Bool"   -> BOOL
   | "Char"   -> CHAR
   | "String" -> STRING
   | "List"   -> LIST
-  | s -> raise (Parse_Exn ("Could not parse type `" ^ s ^ "`."))
+  | _ -> raise (Parse_Exn ("Could not parse type `" ^ s ^ "`."))
 
-let to_string : t -> string = function
+let rec of_string (sexp: Sexp.t) : t =
+  match sexp with
+    | Sexp.(Atom v) -> (of_atomic_string v)
+    | Sexp.(List ([Atom("Array");index;value])) -> ARRAY ((of_string (index)),(of_string (value)))
+    | _ -> raise (Parse_Exn ("Could not parse type `" ^ (Sexp.to_string_hum sexp) ^ "`."))
+    
+
+let rec to_string : t -> string = function
   | INT    -> "Int"
   | BOOL   -> "Bool"
   | CHAR   -> "Char"
   | STRING -> "String"
   | LIST   -> "List"
+  | TVAR (a)    -> a
+  | ARRAY (a,b) -> "(Array" ^ " " ^(to_string a) ^ " " ^ (to_string b) ^ ")"
```

Comment:
Can you

1) align the arrows for the other lines?
2) insert a space before `(to_string a)`?